### PR TITLE
Graveyarding

### PR DESCRIPTION
**Pull request by marcusleemitchell**
Give the ability to archive out of date stories to allow more relevant ones to bubble to the surface.

### CODE REVIEW

**Comment by thejspr** [comment]
Would love to see this merged soon. @mezis any objections? Makes my pending PR ( #203 ) irrelevant

File: .env [hunk 1]
```diff
@@ -1,4 +1,4 @@
-APP_DOMAIN=appfab.dev.io
+APP_DOMAIN=localhost
```
Comment (mezis) [inline .env:1]: That's going to work as an unintented side effect, but I can't take it. HTTPs and domain switching will stop working locally (I'll fork your PR to get it merged though).

File: app/models/ability.rb [hunk 2]
```diff
@@ -62,6 +62,8 @@ def initialize(user)
     end
 
     # Idea lifecycle
+    can :bury,           Idea
+    can :revive,         Idea
```
Comment (mezis) [inline app/models/ability.rb:66]: So everyone can do that then? Doesn't sound too safe.

File: .env (hunk 1 above)
Comment (marcusleemitchell) [inline .env:1]: It's something I can revert if necessary. Do I need POW to get *.dev.io working?

File: app/models/idea_state_machine.rb [hunk 3]
```diff
@@ -56,6 +58,17 @@ def all_state_names
     state :signed_off,   value: 7
     state :live,         value: 8
 
+    event :bury do
+      transition (any - [:implemented, :signed_off, :live, :archived, :graveyarded, :archived]) => :graveyarded
+      transition [:implemented, :signed_off, :live] => :archived
+      # transition :graveyarded => same
+      # transition :archived => same
```
Comment (mezis) [inline app/models/idea_state_machine.rb:65]: Leftovers here.

File: .env (hunk 1 above)
Comment (mezis) [inline .env:1]: You probably do indeed.

Comment (mezis) [inline .env:1]: Don't worry though, I'll take core of wrapping this up.

**Comment by mezis** [comment]
Superseded by the aforementioned PR.

**Pull request closed without merging**